### PR TITLE
Improve forgotten email edge case

### PR DESCRIPTION
**Pull request by AdrienClairembault**
There may be multiple users in GLPI with the same email (there is no unique email constraint).
If one of these users forget his password, the password recovery process fails with "Email address not found" which is confusing as the email address does in fact exist.
This may lead the user (and his administrator) to lose time trying to understand the real issue.

These changes add a different error message for this edge case.

Suggested by @Pieurre

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |

### CODE REVIEW

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]
LGTM. Maybe add tests cases

File: inc/user.class.php [hunk 1]
```diff
@@ -4854,7 +4882,11 @@ public function updateForgottenPassword(array $input) {
          }
 
       } else {
-         throw new ForgetPasswordException(__('Email address not found.'));
+         if (self::countUsersByEmail($input['email'], $condition) > 1) {
+            throw new ForgetPasswordException(__('Multiple users were found for this email address. Please contact your administrator.'));
+         } else {
+            throw new ForgetPasswordException(__('Email address not found.'));
+         }
```
Comment (cedric-anne) [inline inc/user.class.php:4889]: It is generally a bad idea to expose the fact that an account exists for a given email, as it can help to identify login / emails of the user base.
May be the right way to do would be to send the recovery email to all users having the email. The same user would receive then multiple emails, and maybe one of them would contain informations that can help him to identify which recovery link to use.

Comment (trasher) [reply inc/user.class.php:4889]: I'm not sure sending several emails is the solution; it's not sure the related account really belongs to the attached email; and this should also cause issues to enable the wrong account on some cases I guess.

Anyways, I'm OK to stay "vague" in our error messages related to email and authentication. We may add an info in logs; so admin can know what went wrong.

**Comment by AdrienClairembault** [comment]
Done as discussed (log the error and informs the user that he may need to reach out to his admin if if don't receive any emails).

I've removed the email field from the new password form since it's already validated on the previous step.

**Review by cedric-anne — COMMENTED** [review]
Seems OK, but I guess API have to be updated too.

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]
